src/clients/AcrossHubPoolClient.ts

```diff
@@ -0,0 +1,659 @@
+import { assign, Contract, winston, BigNumber, EventSearchConfig, MakeOptional, BigNumberish } from "../utils";
+import {
+  fetchTokenInfo,
+  Event,
+  EventFilter,
+  sortEventsDescending,
+  spreadEvent,
+  spreadEventWithBlockNumber,
+  paginatedEventQuery,
+  toBN,
+} from "../utils";
+import { IGNORED_HUB_EXECUTED_BUNDLES, IGNORED_HUB_PROPOSED_BUNDLES } from "../common";
+import { Deposit, L1Token, CancelledRootBundle, DisputedRootBundle, LpToken } from "../interfaces";
+import { ExecutedRootBundle, PendingRootBundle, ProposedRootBundle } from "../interfaces";
+import { CrossChainContractsSet, DestinationTokenWithBlock, SetPoolRebalanceRoot } from "../interfaces";
+import _ from "lodash";
+
+type _HubPoolUpdate = {
+  success: true;
+  currentTime: number;
+  latestBlockNumber: number;
+  pendingRootBundleProposal: PendingRootBundle;
+  events: Record<string, Event[]>;
+  searchEndBlock: number;
+};
+export type HubPoolUpdate = { success: false } | _HubPoolUpdate;
+
+type HubPoolEvent =
+  | "SetPoolRebalanceRoute"
+  | "SetPoolRebalanceRoute"
+  | "L1TokenEnabledForLiquidityProvision"
+  | "ProposeRootBundle"
+  | "RootBundleCanceled"
+  | "RootBundleDisputed"
+  | "RootBundleExecuted"
+  | "CrossChainContractsSet";
+
+type L1TokensToDestinationTokens = {
+  [l1Token: string]: { [destinationChainId: number]: string };
+};
+export class HubPoolClient {
+  // L1Token -> destinationChainId -> destinationToken
+  private l1TokensToDestinationTokens: L1TokensToDestinationTokens = {};
+  private l1Tokens: L1Token[] = []; // L1Tokens and their associated info.
+  private lpTokens: { [token: string]: LpToken } = {};
+  private proposedRootBundles: ProposedRootBundle[] = [];
+  private canceledRootBundles: CancelledRootBundle[] = [];
+  private disputedRootBundles: DisputedRootBundle[] = [];
+  private executedRootBundles: ExecutedRootBundle[] = [];
+  private crossChainContracts: { [l2ChainId: number]: CrossChainContractsSet[] } = {};
+  private l1TokensToDestinationTokensWithBlock: {
+    [l1Token: string]: { [destinationChainId: number]: DestinationTokenWithBlock[] };
+  } = {};
+  private pendingRootBundle: PendingRootBundle | undefined;
+
+  public isUpdated = false;
+  public firstBlockToSearch: number;
+  public latestBlockNumber: number | undefined;
+  public currentTime: number | undefined;
+
+  constructor(
+    readonly logger: winston.Logger,
+    readonly hubPool: Contract,
+    readonly chainId: number = 1,
+    readonly eventSearchConfig: MakeOptional<EventSearchConfig, "toBlock"> = { fromBlock: 0, maxBlockLookBack: 0 }
+  ) {
```

Comment:
Update the reference to the constant `IGNORED_HUB_PROPOSED_BUNDLES` and `IGNORED_HUB_EXECUTED_BUNDLES` with an overridable instance config variable within the class.